contracts/BufferLib.sol

```diff
@@ -18,16 +18,30 @@ library BufferLib {
   * @param _length How many bytes to read and consume from the buffer.
   * @return A `bytes memory` containing the first `_length` bytes from the buffer, counting from the cursor position.
   */
-  function read(Buffer memory _buffer, uint64 _length) internal pure returns (bytes memory) {
+  function read(Buffer memory _buffer, uint32 _length) internal pure returns (bytes memory) {
```

Comment:
Maybe an explanation in the why an `uint32` is enough. It supports to read up to 537MB which is already...a lot :smile: